src/main/java/build/buildfarm/server/BuildFarmServer.java

```diff
@@ -90,10 +92,15 @@ public BuildFarmServer(
       throws InterruptedException, ConfigurationException {
     super("BuildFarmServer");
 
-    instance = BuildFarmInstances.createInstance(session, config.getInstance(), this::stop);
-
     healthStatusManager = new HealthStatusManager();
+    instance = BuildFarmInstances.createInstance(session, config.getInstance(), this::stop);
+    server = createServer(serverBuilder, instance, config);
+    logger.log(Level.INFO, String.format("%s initialized", session));
+  }
 
+  private Server createServer(
+      ServerBuilder<?> serverBuilder, Instance instance, BuildFarmServerConfig config)
+      throws InterruptedException, ConfigurationException {
```

Comment:
i would do these (and other refactoring below) in a separate diff.

Reply:
reverted createServer function.  A little more clear now.